flink-runtime/src/main/java/org/apache/flink/runtime/executiongraph/SpeculativeExecutionVertex.java

```diff
@@ -46,6 +48,8 @@ public class SpeculativeExecutionVertex extends ExecutionVertex {
 
     private int originalAttemptNumber;
 
+    final Map<Integer, Integer> nextInputSplitIndexToConsumeByAttempts;
```

Comment:
This map should be clear when `resetForNewExecution`

Reply:
Good catch! We should do it to avoid leaks. 
And it's also better to remove the attempt from `nextInputSplitIndexToConsumeByAttempts` in `archiveFailedExecution`. 
I will make these two changes.